src/manager/node.c

```diff
@@ -1126,6 +1142,10 @@ AgentRequest *node_request_list_units(
 static int method_list_units_callback(AgentRequest *req, sd_bus_message *m, UNUSED sd_bus_error *ret_error) {
         sd_bus_message *request_message = req->userdata;
 
+        if (req->is_cancelled) {
```

Comment:
None of these are necessary if you synthesize a message error in the cancel callback, the below sd_bus_message_is_method_error() call will be enough.

However, you might want node_method_list_units() and the other "start a request" methods to return directly if the node is shutdown, to avoid creating new request references.

Reply:
Done